examples/models/object_detection/faster_rcnn/train.py

```diff
@@ -18,16 +18,38 @@
 Last modified: 2022/09/27
 Description: Use KerasCV to train a RetinaNet on Pascal VOC 2007.
 """
-from datetime import datetime
+import sys
 
 import tensorflow as tf
 import tensorflow_datasets as tfds
+from absl import flags
 
 import keras_cv
 
+flags.DEFINE_string(
+    "weights_path",
+    "weights_{epoch:02d}.h5",
+    "Directory which will be used to store weight checkpoints.",
+)
+flags.DEFINE_string(
+    "tensorboard_path",
+    "logs",
+    "Directory which will be used to store tensorboard logs.",
+)
+FLAGS = flags.FLAGS
+FLAGS(sys.argv)
+
 # parameters from FasterRCNN [paper](https://arxiv.org/pdf/1506.01497.pdf)
 
-strategy = tf.distribute.MirroredStrategy()
+# Try to detect an available TPU. If none is present, default to MirroredStrategy
+try:
+    tpu = tf.distribute.cluster_resolver.TPUClusterResolver.connect()
+    strategy = tf.distribute.TPUStrategy(tpu)
+except ValueError:
+    # MirroredStrategy is best for a single machine with one or multiple GPUs
+    strategy = tf.distribute.MirroredStrategy()
+print("Number of accelerators: ", strategy.num_replicas_in_sync)
+
```

Comment:
It would be better to have a `keras.utility` function to automate the device selection.

Reply:
To me this sounds more like a `tf.distribute.auto_select_strategy` :-)

Reply:
Having a trainer object that takes care of runtime args is definitely not a bad idea

Reply:
```
trainer = Trainer(devices=8, accelerator="gpu")
```

isn't this just `MirroredStrategy(devices=8)`

Reply:
@LukeWood you're right. Now, the following code is widely used. 

```python
try:
    tpu = tf.distribute.cluster_resolver.TPUClusterResolver.connect()
    strategy = tf.distribute.TPUStrategy(tpu)
except ValueError:
    # MirroredStrategy is best for a single machine with one or multiple GPUs
    strategy = tf.distribute.MirroredStrategy()
print("Number of accelerators: ", strategy.num_replicas_in_sync)
```

That's why I was thinking it might be good to have a simple utility function to automate it, simple and clean.

i.e. `keras.utils.set_strategy(devices=2, accelerator="gpu")`